src/mongocxx/test/collection.cpp

```diff
@@ -388,4 +393,57 @@ TEST_CASE("CRUD functionality", "[driver::collection]") {
 
         auto results = coll.aggregate(p);
     }
+
+    SECTION("distinct works", "[collection]") {
+        using bsoncxx::builder::stream::document;
+
+        auto distinct_cname = "distinct_coll";
+        auto distinct_coll = db[distinct_cname];
+        if (db.has_collection(distinct_cname)) {
+            distinct_coll.drop();
+        }
+
+        document doc1{};
+        doc1 << "foo" << "baz" << "garply" << 1;
+        document doc2{};
+        doc2 << "foo" << "bar" << "garply" << 2;
+        document doc3{};
+        doc3 << "foo" << "baz" << "garply" << 2;
+        document doc4{};
+        doc4 << "foo" << "quux" << "garply" << 9;
+
+        bulk_write bulk{false /* unordered */};
```

Comment:
Off topic: Do we really just use raw bool's for the ordered-ness of a bulk_write? That makes me sad. Does CRUD spec allow us leeway here?